ArcMovies/MoviesScene/CollectionView/Cells/MoviePosterCell/MoviePosterCell.swift

```diff
@@ -0,0 +1,220 @@
+    }
+    
+}
+
+// MARK: Lifecycle
+
+extension MoviePosterCell {
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        // Initialization code
+        commonInit()
+    }
+
+    private func commonInit() {
+        configureViews()
+    }
+}
+
+// MARK: Configs
+
+extension MoviePosterCell {
+    
+    private func setPosterImage() {
+        // Start loader feedback
+        activityIndicator.startAnimating()
+        
+        // Load the poster url
+        Poster.setImageURL(url: viewModel?.posterURL) { [weak self] in
+            
+            // Stop de loader when completes
+            self?.activityIndicator.stopAnimating()
+            
+            //Show Poster animated
+            self?.showPoster()
+            
+            // Configure colors
+            self?.defineColors(with: self?.Poster.image)
+        }
+    }
+    
+    private func showPoster() {
+        DispatchQueue.main.async {
+            UIView.animate(withDuration: 0.2) { [weak self] in
+                self?.Poster.alpha = 1
+                self?.layoutIfNeeded()
+            }
+        }
+    }
+    
+    private func showShadow(_ shadow : UIView?) {
+        DispatchQueue.main.async {
+            UIView.animate(withDuration: 0.2) { [weak self] in
+                shadow?.alpha = 1
+                self?.layoutIfNeeded()
+            }
+        }
+    }
+
+    private func configureViews() {
+        // Config. poster container
+        PosterContainer.layer.masksToBounds = true
+        PosterContainer.layer.cornerRadius = cornerRadius
+        PosterContainer.alpha = 1
+        
+        // Config. self
+        contentView.layer.masksToBounds = false
+        layer.masksToBounds = false
+    }
+    
+    private func configureShadow() {
+        guard self.shadowView == nil else {
+            // Show shadow
+            showShadow(self.shadowView)
+            return
+        }
+        var shadowView = UIView(frame: .zero)
+
+        // Set poster shadow
+        With(&shadowView) {
+            $0.alpha = 0
+            $0.backgroundColor = UIColor(white: 0, alpha: 0)
+            $0.layer.masksToBounds = false
+            $0.layer.shadowColor = UIColor(white: 0, alpha: 0).cgColor
+            $0.layer.shadowRadius = 25
+            $0.layer.shadowOpacity = 0.5
+            $0.layer.shadowOffset = CGSize(width: 0, height: 8)
+            $0.layer.shouldRasterize = true
+            $0.layer.shadowPath = UIBezierPath(roundedRect: CGRect(x: 0, y: 0, width: bounds.width, height: bounds.height), cornerRadius: PosterContainer.layer.cornerRadius).cgPath
+        }
+        addShowViewWithConstraint(shadowView)
+    }
+    
+    /// Define views colors
+    private func defineColors(with image: UIImage?) {
+        guard let image = image else { return }
+        DispatchQueue.main.async {
+            image.getColors(quality: .low) { [weak self] colors in
+                self?.setShadowColor(with: colors.background.cgColor)
```

Comment:
Is there a leak on self here? Why?

Reply:
@jaikme

Reply:
No its not. besides this method of ```UIImageColors``` execute on a global concurrent queue, it does not leak. I can remove the [weak self] here, but I prefer let it there, in case I have a memory leak, and will know its not it, so I don't need check it again.Leaving it there does not cause any problems.

The DispatchQueue here proved that it gives a better up update for what I need.